basic_system/src/system_implementation/flat_storage_model/account_cache.rs

```diff
@@ -895,6 +904,128 @@ where
+            )
+        } else {
+            // Bytecode is: 0xef0100 || address
+            let mut code = [0u8; 23];
+            code[0..3].copy_from_slice(&EIP7702_DELEGATION_MARKER);
+            code[3..].copy_from_slice(&delegate.to_be_bytes::<{ B160::BYTES }>());
+
+            // compute observable and true hashes of bytecode
+            let observable_bytecode_hash = {
+                let native_cost = keccak256_native_cost::<R>(code.len());
+                resources.charge(&R::from_native(native_cost))?;
+                use crypto::sha3::Keccak256;
+                use crypto::MiniDigest;
+                let digest = Keccak256::digest(code);
+                Bytes32::from_array(digest)
+            };
+
+            let observable_bytecode_len = code.len() as u32;
+
+            // We compute bytecode hash including padding, for compatibility
+            // We set EE type to EVM, just to use Blake in the helper function
+            let bytecode_hash =
+                Self::compute_bytecode_hash(ExecutionEnvironmentType::EVM, &code, &[], resources)?;
+            let artifacts_len = 0;
+            let padding_len = bytecode_padding_len(code.len());
+            let bytecode_len = observable_bytecode_len + (padding_len as u32) + artifacts_len;
+            let padding = [0u8; core::mem::size_of::<u64>() - 1];
+            let padding = &padding[..padding_len];
+            // save bytecode
+            preimages_cache.record_preimage::<PROOF_ENV>(
+                ExecutionEnvironmentType::NoEE,
+                &(PreimageRequest {
+                    hash: bytecode_hash,
+                    expected_preimage_len_in_bytes: bytecode_len,
+                    preimage_type: PreimageType::Bytecode,
+                }),
+                resources,
+                &[&code, padding, &[]],
+            )?;
+            (
+                observable_bytecode_hash,
+                observable_bytecode_len,
+                bytecode_hash,
+                artifacts_len,
+                evm_interpreter::ARTIFACTS_CACHING_CODE_VERSION_BYTE,
+                true,
+            )
+        };
+
+        resources.charge(&R::from_native(R::Native::from_computational(
+            WARM_ACCOUNT_CACHE_WRITE_EXTRA_NATIVE_COST,
+        )))?;
+
+        account_data.update(|cache_record| {
+            cache_record.update(|v, m| {
+                v.observable_bytecode_hash = observable_bytecode_hash;
+                v.observable_bytecode_len = observable_bytecode_len;
+                v.bytecode_hash = bytecode_hash;
+                v.unpadded_code_len = observable_bytecode_len;
+                v.artifacts_len = artifacts_len;
+
+                if delegated {
+                    v.versioning_data.set_as_delegated();
+                    // We set EVM as EE, as it's the only EE supported currently.
+                    v.versioning_data
+                        .set_ee_version(ExecutionEnvironmentType::EVM_EE_BYTE);
```

Comment:
Error-prone. Also I guess we discussed, that we need to actually read delegation target, right?

Reply:
Yeah, makes sense, updated it already

Reply:
Can you pls add a note that it doesn't mean that we can't delegate to other EEs? It's only about "placeholder" delegated account version